tests/ha/ha_gnmi_utils.py

```diff
@@ -0,0 +1,365 @@
+    if len(update_list) > 0:
+        cmd += '-m set-update '
+    elif len(delete_list) > 0:
+        cmd += '-m set-delete '
+    elif len(replace_list):
+        cmd += '-m set-replace '
+    else:
+        logger.info("PTF GNMI no items to operate on")
+        return
+    xpath = ''
+    xvalue = ''
+    for path in delete_list:
+        path = path.replace('sonic-db:', '')
+        xpath += ' ' + path
+        xvalue += ' ""'
+    for update in update_list:
+        update = update.replace('sonic-db:', '')
+        result = update.rsplit(':', 1)
+        xpath += ' ' + result[0]
+        xvalue += ' ' + result[1]
+    for replace in replace_list:
+        replace = replace.replace('sonic-db:', '')
+        result = replace.rsplit(':', 1)
+        xpath += ' ' + result[0]
+        if '#' in result[1]:
+            xvalue += ' ""'
+        else:
+            xvalue += ' ' + result[1]
+    cmd += '--xpath ' + xpath
+    cmd += ' '
+    cmd += '--value ' + xvalue
+    logger.info(f"PTF GNMI command: {cmd}")
+    output = ptfhost.shell(cmd, module_ignore_errors=True)
+    error = "GRPC error\n"
+    if error in output['stdout']:
+        result = output['stdout'].split(error, 1)
+        raise Exception("GRPC error:" + result[1])
+    return
+
+
+def gnmi_get(duthost, ptfhost, path_list):
+    """
+    Send GNMI get request with GNMI client
+
+    Args:
+        duthost: fixture for duthost
+        ptfhost: fixture for ptfhost
+        path_list: list for get path
+
+    Returns:
+        msg_list: list for get result
+    """
+    env = GNMIEnvironment(duthost)
+    ip = duthost.mgmt_ip
+    port = env.gnmi_port
+    cmd = 'python /root/gnxi/gnmi_cli_py/py_gnmicli.py '
+    cmd += '--timeout 30 '
+    cmd += '-t %s -p %u ' % (ip, port)
+    cmd += '-xo sonic-db '
+    cmd += '-rcert /root/%s ' % (env.gnmi_ca_cert)
+    cmd += '-pkey /root/%s ' % (env.gnmi_client_key)
+    cmd += '-cchain /root/%s ' % (env.gnmi_client_cert)
+    cmd += '--encoding 4 '
+    cmd += '-m get '
+    cmd += '--xpath '
+    for path in path_list:
+        path = path.replace('sonic-db:', '')
+        cmd += " " + path
+    output = ptfhost.shell(cmd, module_ignore_errors=True)
+    msg = output['stdout'].replace('\\', '')
+    error = "GRPC error\n"
+    if error in msg:
+        result = msg.split(error, 1)
+        raise Exception("GRPC error:" + result[1])
+    mark = 'The GetResponse is below\n' + '-'*25 + '\n'
+    if mark in msg:
+        result = msg.split(mark, 1)
+        msg_list = result[1].split('-'*25)[0:-1]
+        return [msg.strip("\n") for msg in msg_list]
+    else:
+        raise Exception("error:" + msg)
+
+
+def apply_messages(
+    localhost,
+    duthost,
+    ptfhost,
+    messages,
+    dpu_index,
+    setup_ha_config,
+    set_db=True,
+    wait_after_apply=5,
+    max_updates_in_single_cmd=1024,
+):
+    env = GNMIEnvironment(duthost)
+    update_list = []
+    delete_list = []
+    gnmi_key = "DASH_HA_SET_CONFIG_TABLE[key=HaSet1]"
```

Comment:
We also have `DASH_HA_GLOBAL_CONFIG` and `DASH_HA_SCOPE_CONFIG_TABLE` and etc. to cover?

Can we make `apply_messages` method more generic?